lib/Sema/TypeCheckDecl.cpp

```diff
@@ -2175,14 +2223,16 @@ static void checkAccessControl(TypeChecker &TC, const Decl *D) {
 
   case DeclKind::Struct: {
     auto SD = cast<StructDecl>(D);
-    checkGenericParamAccess(TC, SD->getGenericParams(), SD);
+    checkGenericParamAccess(TC, SD->getGenericParams(), SD,
+                            /*checkUsableFromInline=*/false);
```

Comment:
(here and elsewhere) Why is this `false` and not inherited? Can you leave a comment to that effect?

Reply:
That's correct. `checkAccessControl()` only checks access control rules. The separate `checkUsableFromInline()` performs the same visitation of all TypeLocs, calling checkTypeAccess() with `checkUsableFromInline=true`.